### PR TITLE
tests/Redpanda: skip log scan if the file dne

### PR DESCRIPTION
**Pull request by NyaliaLui**
## Cover letter

This commit adds a flag to RedpandaService that, when set to True,
the service will skip the log scan in raise_on_bad_logs. The default
is to continue the log scan.

## UX changes
* none

## Release notes
* none

### CODE REVIEW

File: tests/rptest/services/redpanda.py [hunk 1]
```diff
@@ -1078,6 +1081,14 @@ def raise_on_bad_logs(self, allow_list=None):
 
         bad_lines = collections.defaultdict(list)
         for node in self.nodes:
+
+            if self._skip_if_no_redpanda_log and not node.account.exists(
```
Comment (BenPope) [inline tests/rptest/services/redpanda.py:1085]: Should we skip it or should we check a particular error message?

Comment (NyaliaLui) [reply tests/rptest/services/redpanda.py:1085]: The error message "/var/lib/redpanda/redpanda.log: No such file or directory" comes from bash not from the redpanda.log file so `redpanda.search_log()` is not applicable here.

Sure I could look for the error message from the grep call on line 1096 but, in my mind, that would also result in an early return.

TLDR -- I don't see a difference.

Comment (BenPope) [reply tests/rptest/services/redpanda.py:1085]: Yeah, this is fine. It's merged into the PR, thanks

**Comment by NyaliaLui** [comment]
Closing because this was already merged.
